docs/blog/posts/af_xdp-dp-integration.md

```diff
@@ -0,0 +1,114 @@
+for redirecting incoming packets to AF_XDP sockets.
+
+#### Integration benefits
+
+So what are the benefits of bpfman integration for the AF_XDP DP and CNI?
+
+- Removes eBPF code from the AF_XDP DP and CNI codebase.
+  - This presented a difficulty particularly when trying to find/update appropriate
+    base container images to use for the AF_XDP device plugin. Different images
+    supported different versions of libbpf or libxdp which meant we kept having to make
+    changes around the loading and attaching of the base eBPF program.
+  - Additionally the CNI runs as a binary on the Kubernetes node so we would need to
+    statically compile libbpf/libxdp as part of the CNI.
+- More diverse XDP program support through bpfmans eBPF Bytecode Image Support. Not
+  only do the AF_XDP eBPF programs no longer need to be stored in the Device Plugin
+  itself, but it's now configurable on a per pool basis.
+- No longer required to leverage Hostpath volume mounts to mount the AF_XDP maps inside
+  a pod. But rather take advantage of the bpfman csi support to ensure that maps are
+  pinned in the context of the Pod itself and not in a bpffs on the host (then shared
+  to the pod).
+
+#### AF_XDP Device Plugin tasks
+
+- Creates a resource pool of netdevs that can be used with AF_XDP.
+- **Prior to bpfman integration**:
+  - Loads the default AF_XDP BPF prog onto the netdev at pod creation and manages info
+    re the xsk_map for that netdev.
+  - Shares the xsk_map file descriptor via UDS (involves a cumbersome handshake with the pod).
+- **After bpfman integration**:
+  - Uses bpfman's client APIs to load the BPF prog.
+  - Shares the xsk_map file descriptor via pinned map with the Pod.
+
+#### AF_XDP CNI tasks
+
+- Moves a netdev from the Host network namespace to the Pod network namespace.
+- Maintains the netdev name (as a unique id across namespaces).
+- Programs/clears netfilters on the netdevice
+- **Prior to bpfman integration**:
+  - Unloads the BPF program when a device is returned to the Host network namespace
+- **After bpfman integration**:
+  - Uses grpc to signal to the Device Plugin to request bpfman to unload the BPF prog.
+
+## Is there a working example?
+
+The bpfman integration with the AF_XDP Device Plugin and CNI was demo'ed as part
+of a series of demos that show the migration of a DPDK application to AF_XDP (without)
+any application modification. The demo can be watched below:
+
+[<img src="https://img.youtube.com/vi/wcmSO5HwNJQ/hqdefault.jpg" width="600" height="300"
+/>](https://www.youtube.com/embed/wcmSO5HwNJQ)
+
+## AF_XDP DP and CNI's integration with bpfman in images
+
+The following sections will present the evolution of the AF_XDP DP and CNI
+from independent bpf program management to leveraging bpfman to manage eBPF programs on
+their behalf.
+
+### AF_XDP DP and CNI managing eBPF programs independently
+
+![AF_XDP DP and CNI managing eBPF programs independently](./img/2024-02-27/af_xdp.png)
+
```

Comment:
I can call out the steps in detail here, WDYT?

Reply:
It think that makes sense!! Most folks reading will come with much less context on what af_xdp actually is so feel free to talk more details!

Reply:
And/Or highlight the steps where bpfman made things better, like Step 9 handshake no longer needed.